src/definitions/device_engagement.rs

```diff
@@ -246,25 +277,36 @@ impl DeviceRetrievalMethod {
     }
 }
 
-impl TryFrom<CborValue> for DeviceRetrievalMethod {
+impl TryFrom<ciborium::Value> for DeviceRetrievalMethod {
     type Error = Error;
-    fn try_from(value: CborValue) -> Result<Self, Self::Error> {
-        if let CborValue::Array(list) = value {
-            let method: [CborValue; 3] = list.try_into().map_err(|_| Error::Malformed)?;
-            match method {
-                [CborValue::Integer(1), CborValue::Integer(1), methods] => {
-                    let nfc_options = NfcOptions::try_from(methods)?;
+    fn try_from(value: ciborium::Value) -> Result<Self, Self::Error> {
+        if let ciborium::Value::Array(list) = value {
+            match list.as_slice() {
+                [ciborium::Value::Integer(i1), ciborium::Value::Integer(i11), methods]
+                    if <ciborium::value::Integer as Into<i128>>::into(*i1) == 1
+                        && <ciborium::value::Integer as Into<i128>>::into(*i11) == 1 =>
+                {
+                    let nfc_options =
+                        NfcOptions::try_from(methods.clone()).map_err(|_| Error::Malformed)?;
```

Comment:
Do we need to map the error? Aren't they already the correct type?
```suggestion
                        NfcOptions::try_from(methods.clone())?;
```

Reply:
changed